tests/system.rs

```diff
@@ -66,27 +85,64 @@ fn print_mismatches(result: HashMap<String, String>) {
 static HANDLE_RESULT: &'static Fn(HashMap<String, String>) = &handle_result;
 
 pub fn idempotent_check(filename: String) -> Result<(), HashMap<String, String>> {
+    let config = get_config(&filename);
     let args = vec!["rustfmt".to_owned(), filename];
-    let mut def_config_file = fs::File::open("default.toml").unwrap();
-    let mut def_config = String::new();
-    def_config_file.read_to_string(&mut def_config).unwrap();
     // this thread is not used for concurrency, but rather to workaround the issue that the passed
     // function handle needs to have static lifetime. Instead of using a global RefCell, we use
     // panic to return a result in case of failure. This has the advantage of smoothing the road to
     // multithreaded rustfmt
     thread::catch_panic(move || {
-        run(args, WriteMode::Return(HANDLE_RESULT), &def_config);
+        run(args, WriteMode::Return(HANDLE_RESULT), &config);
```

Comment:
you can replace the `unwrap()` with an expect, but that won't yield the inner error message. Instead this would probably need some more extensive downcast chaining for different result types